src/txkube/test/test_authentication.py

```diff
@@ -68,6 +73,37 @@
 -----END CERTIFICATE-----
 """
 
+try:
+    from twisted.internet.address import IPv4Address
+    from twisted.internet.interfaces import IReactorPluggableNameResolver
+
+    class Resolution(object):
+        name = None
+
+    class _FakeResolver(object):
+        def resolveHostName(self, resolutionReceiver, hostName, *args,  **kwargs):
+            portNumber = kwargs.pop('portNumber')
+            r = Resolution()
+            r.name = hostName
+
+            resolutionReceiver.resolutionBegan(r)
+            resolutionReceiver.addressResolved(IPv4Address('TCP', '1.2.3.4', portNumber))
+            resolutionReceiver.resolutionComplete()
+            return r
+
+    @implementer(IReactorPluggableNameResolver)
+    class _ResolvingMemoryReactor(MemoryReactor):
+        nameResolver = _FakeResolver()
+
+    MemoryReactor = _ResolvingMemoryReactor
+    expectedTCPConnectResult = '1.2.3.4'
```

Comment:
Oughtn't this clearly be part of the public testing interface published by Twisted itself?